lib/discourse_ip_info.rb

```diff
@@ -0,0 +1,39 @@
+require_dependency 'maxminddb'
+
+class DiscourseIpInfo
+  include Singleton
+
+  def initialize
+    begin
+      @mmdb_filename = File.join(Rails.root, 'vendor', 'data', 'GeoLite2-City.mmdb')
+      @mmdb = MaxMindDB.new(@mmdb_filename, MaxMindDB::LOW_MEMORY_FILE_READER)
+    rescue
```

Comment:
This is too broad. What specific exceptions are you trying to rescue?

Reply:
There are plenty of exceptions which should be rescued and are not very intuitive. Some of those are:

- `Errno::ENOENT` if file does not exist;
- `NoMethodError` if the MaxMindDB file is empty (0 bytes, caused by calling `rindex` on `nil`);
- `RuntimeError` if the MaxMindDB is invalid (raised by `maxminddb` gem).

I would not like MaxMindDb to cause any problem if `maxminddb` causes any issues so that's why I adopted a 'rescue-all' attitude.